src/@context/Automation/AutomationProvider.tsx

```diff
@@ -0,0 +1,239 @@
+import { LoggerInstance } from '@oceanprotocol/lib'
+import { Wallet, ethers } from 'ethers'
+import React, {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState
+} from 'react'
+import { useAccount, useBalance, useChainId, useProvider } from 'wagmi'
+import { accountTruncate, getTokenBalance } from '../../@utils/wallet'
+import { useUserPreferences } from '../UserPreferences'
+import { toast } from 'react-toastify'
+import Modal from '../../components/@shared/atoms/Modal'
+import Button from '../../components/@shared/atoms/Button'
+import styles from './AutomationProvider.module.css'
+import Loader from '../../components/@shared/atoms/Loader'
+import { useMarketMetadata } from '../MarketMetadata'
+import DeleteAutomationModal from './DeleteAutomationModal'
+
+export interface NativeTokenBalance {
+  symbol: string
+  balance: string
+}
+export interface AutomationProviderValue {
+  autoWallet: Wallet
+  autoWalletAddress: string
+  isAutomationEnabled: boolean
+  balance: UserBalance
+  nativeBalance: NativeTokenBalance
+  isLoading: boolean
+  decryptPercentage: number
+  updateBalance: () => Promise<void>
+  setIsAutomationEnabled: (isEnabled: boolean) => void
+  deleteCurrentAutomationWallet: () => void
+  importAutomationWallet: (encryptedJson: string) => Promise<boolean>
+  hasValidEncryptedWallet: () => boolean
+  decryptAutomationWallet: (password: string) => Promise<boolean>
+}
+
+// Refresh interval for balance retrieve - 20 sec
+const refreshInterval = 20000
```

Comment:
thought: The `refreshInterval` is set to 20 seconds. Depending on the number of users and the scale of the application, this might result in a lot of requests. We should consider this in our architecture planning (focus on the nodes) as our user base grows.

Reply:
Agreed. We also have another balance call on the portal for the connected wallet (e.g., Metamask) where we also use a interval (20s iirc)